app/queries/get_incomplete_reference_applications_ready_to_nudge.rb

```diff
@@ -1,49 +1,35 @@
 class GetIncompleteReferenceApplicationsReadyToNudge
-  MAILER = 'candidate_mailer'.freeze
-  MAIL_TEMPLATE = 'nudge_unsubmitted_with_incomplete_references'.freeze
-  COMMON_COMPLETION_ATTRS = GetUnsubmittedApplicationsReadyToNudge::COMMON_COMPLETION_ATTRS
+  COMMON_COMPLETION_ATTRS_WITHOUT_REFERENCES = GetUnsubmittedApplicationsReadyToNudge::COMMON_COMPLETION_ATTRS.filter do |attr|
+    attr != 'references_completed'
+  end
 
   def call
     uk_and_irish_names = NATIONALITIES.select do |code, _name|
       code.in?(ApplicationForm::BRITISH_OR_IRISH_NATIONALITIES)
     end.map(&:second)
-    uk_and_irish = uk_and_irish_names.map { |name| ActiveRecord::Base.connection.quote(name) }.join(',')
 
     ApplicationForm
+      # Only candidates with some unsubmitted choices
+      .joins(:application_choices)
       .joins(:candidate)
-      .where.not('candidate.submission_blocked': true)
-      .where.not('candidate.account_locked': true)
-      .where.not('candidate.unsubscribed_from_emails': true)
+      .joins("LEFT OUTER JOIN emails ON emails.application_form_id = application_forms.id AND emails.mailer = 'candidate_mailer' AND emails.mail_template = 'nudge_unsubmitted_with_incomplete_references'")
+      .joins('LEFT OUTER JOIN application_choices ac_primary ON ac_primary.application_form_id = application_forms.id')
+      .joins('LEFT OUTER JOIN course_options ON course_options.id = ac_primary.course_option_id')
+      .joins("LEFT OUTER JOIN courses courses_primary ON courses_primary.id = course_options.course_id AND LOWER(courses_primary.level) = 'primary'")
```

Comment:
Because of all these joins should we call distinct? Will we have problems with duplicate results and therefore duplicate emails being sent?

Reply:
One example: An application form with two unsubmitted application choices with incomplete references. Can we add one spec for this, please?

Reply:
Yes, we should call distinct. Will add spec.